### PR TITLE
Bump plutus

### PR DESCRIPTION
**Pull request by luigy**

### CODE REVIEW

**Comment by silky** [comment]
@luigy i think we should start marrying the version of plutus to the particular tag on the plutus repo; such as `plutus-starter-devcontainer/v1.0.5`.

this will help because:

1. we'll be forced to make such a tag! this is great, because doing so pushes up a new devcontainer image ( https://github.com/input-output-hk/plutus/blob/master/.buildkite/pipeline.yml#L29 )
2. it'll make it so we can't be out of sync here

i'll give this a try for these changes once [this pr](https://github.com/input-output-hk/plutus/pull/3374#pullrequestreview-684154696) is merged.

@michaelpj does this seem like a good idea to you?

File: nix/sources.json [hunk 1]
```diff
@@ -5,10 +5,10 @@
         "homepage": "",
         "owner": "input-output-hk",
         "repo": "plutus",
-        "rev": "5e20b2df3ac55b55b9d70eb7271b295ac797ed5e",
-        "sha256": "02gsyxrkm79dh1diqgh89gdxm3f4vpk8f9k6ckhnj7kjmpyi5r5s",
+        "rev": "7c2a01f3c94b296fda377c63bf8fab582d82498e",
+        "sha256": "0sw2w8wpbnxjxnhal281byg9xq7b5dwlcfv02mdfs7jyg6ia3h9i",
         "type": "tarball",
-        "url": "https://github.com/input-output-hk/plutus/archive/5e20b2df3ac55b55b9d70eb7271b295ac797ed5e.tar.gz",
+        "url": "https://github.com/input-output-hk/plutus/archive/7c2a01f3c94b296fda377c63bf8fab582d82498e.tar.gz",
```
Comment (silky) [inline nix/sources.json:11]:
```suggestion
        "url": "https://github.com/input-output-hk/plutus/archive/plutus-starter-devcontainer/v1.0.6.tar.gz",
```

File: nix/sources.json [hunk 2]
```diff
@@ -5,10 +5,10 @@
         "homepage": "",
         "owner": "input-output-hk",
         "repo": "plutus",
-        "rev": "5e20b2df3ac55b55b9d70eb7271b295ac797ed5e",
-        "sha256": "02gsyxrkm79dh1diqgh89gdxm3f4vpk8f9k6ckhnj7kjmpyi5r5s",
+        "rev": "7c2a01f3c94b296fda377c63bf8fab582d82498e",
+        "sha256": "0sw2w8wpbnxjxnhal281byg9xq7b5dwlcfv02mdfs7jyg6ia3h9i",
```
Comment (silky) [inline nix/sources.json:9]:
```suggestion
        "sha256": "1jzbcsdrv0b43dj7bwbd1fbk71f7gph6zzb8y29n9cn3j8illnyc",
```

File: nix/sources.json [hunk 3]
```diff
@@ -5,10 +5,10 @@
         "homepage": "",
         "owner": "input-output-hk",
         "repo": "plutus",
-        "rev": "5e20b2df3ac55b55b9d70eb7271b295ac797ed5e",
-        "sha256": "02gsyxrkm79dh1diqgh89gdxm3f4vpk8f9k6ckhnj7kjmpyi5r5s",
+        "rev": "7c2a01f3c94b296fda377c63bf8fab582d82498e",
```
Comment (silky) [inline nix/sources.json:8]:
```suggestion
        "rev": "plutus-starter-devcontainer/v1.0.6",
```

File: nix/pkgs/haskell/haskell.nix [hunk 4]
```diff
@@ -18,7 +18,7 @@ let
 
     sha256map = {
       "https://github.com/Quid2/flat.git"."95e5d7488451e43062ca84d5376b3adcc465f1cd" = "06l31x3y93rjpryvlxnpsyq2zyxvb0z6lik6yq2fvh36i5zwvwa3";
-      "https://github.com/input-output-hk/plutus.git"."5e20b2df3ac55b55b9d70eb7271b295ac797ed5e" = "02gsyxrkm79dh1diqgh89gdxm3f4vpk8f9k6ckhnj7kjmpyi5r5s";
+      "https://github.com/input-output-hk/plutus.git"."7c2a01f3c94b296fda377c63bf8fab582d82498e" = "0sw2w8wpbnxjxnhal281byg9xq7b5dwlcfv02mdfs7jyg6ia3h9i";
```
Comment (silky) [inline nix/pkgs/haskell/haskell.nix:21]:
```suggestion
      "https://github.com/input-output-hk/plutus.git"."plutus-starter-devcontainer/v1.0.6" = "1jzbcsdrv0b43dj7bwbd1fbk71f7gph6zzb8y29n9cn3j8illnyc";
```

File: cabal.project [hunk 5]
```diff
@@ -27,7 +27,7 @@ source-repository-package
     prettyprinter-configurable
     quickcheck-dynamic
     word-array
-  tag: 5e20b2df3ac55b55b9d70eb7271b295ac797ed5e
+  tag: 7c2a01f3c94b296fda377c63bf8fab582d82498e
```
Comment (silky) [inline cabal.project:30]: I've pushed a new tag, so give this a try:

```suggestion
  tag: plutus-starter-devcontainer/v1.0.6
```

**Comment by silky** [comment]
I've pushed a new tag so that this will build properly against that tag instead of the commit

There's one more thing to change, for good measure: in `./devcontainer/devcontainer.json`, we should always bump and pin the tagged version (so here, the diff is:)

```
-    "image": "docker.io/inputoutput/plutus-starter-devcontainer:latest",
+    "image": "docker.io/inputoutput/plutus-starter-devcontainer:v1.0.6",
```

this is because vscode, in it's infinite wisdom, doesn't seem to check that it's version matches 'latest', and it doesn't even seem to me to have an easy option to force pulling it!

**Comment by michaelpj** [comment]
Sounds fine to me. At some point we'll need to start doing tagged releases of the Plutus Applications libraries: then we can switch to using those tags.

**Comment by silky** [comment]
@luigy sorry for the nit-pick; but this is good to go with the addition of the change to the `./devcontainer/devcontainer.json` file in my above comment 😀  ( i didn't know how to make it a suggested change because it wasn't in a file in the diff! )

**Comment by luigy** [comment]
@silky thanks for reminding me. I was actually testing that change earlier and forgot to push :)

**Comment by luigy** [comment]
though  vscode devcontainer doesn't seem to use that tag and based on `docker ps` output it insists on using `latest` even after wiping `~/.vscode` and `docker system prune -a` though I'm not familiar with docker at all so I may me missing something obvious here

**Comment by silky** [comment]
@luigy that's very strange indeed ...

also, i can't see why the hyrda-eval has failed? it seems like it worked??

**Comment by koslambrou** [comment]
@luigy @silky Yeah, this also happens to me (I think it also happens in the `plutus` repo). The problem is that when you push a commit which doesn't change anything in the code, the hydra server doesn't rerun the CI, and returns a failed state.

Try to commit something which modifies the code in `Game.hs` for example.

**Pull request merged**